imageio/plugins/pyav.py

```diff
@@ -0,0 +1,839 @@
+
+        Notes
+        -----
+        The provided metadata provides information about the ImageResource and
+        does not include modifications by any filters (through
+        ``filter_sequence`` or ``filter_graph``).
+
+        """
+
+        video_width = self._video_stream.codec_context.width
+        video_height = self._video_stream.codec_context.height
+        pix_format = format or self._video_stream.codec_context.pix_fmt
+        frame_template = av.VideoFrame(video_width, video_height, pix_format)
+
+        shape = _get_frame_shape(frame_template)
+        if index is None:
+            n_frames = self._video_stream.frames
+            shape = (n_frames,) + shape
+
+        return ImageProperties(
+            shape=tuple(shape),
+            dtype=_format_to_dtype(frame_template.format),
+            is_batch=True if index is None else False,
+        )
+
+    def metadata(
+        self,
+        index: int = None,
+        exclude_applied: bool = True,
+        constant_framerate: bool = True,
+    ) -> Dict[str, Any]:
+        """Read a dictionary filled with metadata.
+
+        Parameters
+        ----------
+        index : int
+            If None (default) return global metadata (the metadata stored in the
+            container and video stream). If not None, return the side data
+            stored in the frame at the given index.
+        exclude_applied : bool
+            Currently, this parameter has no effect. It exists for compliance with
+            the ImageIO v3 API.
+        constant_framerate : bool
+            If True (default) assume the video's framerate is constant. This
+            allows for faster seeking in side the file. If False, the video
+            is reset before each read and searched from the beginning.
+
+        Returns
+        -------
+        metadata : dict
+            A dictionary filled with format-specific metadata fields and their
+            values.
+
+        """
+
+        metadata = dict()
+
+        if index is None:
+            # useful flags defined on the container and/or video stream
+            metadata.update(
+                {
+                    "video_format": self._video_stream.codec_context.pix_fmt,
+                    "codec": self._video_stream.codec.name,
+                    "long_codec": self._video_stream.codec.long_name,
+                    "profile": self._video_stream.profile,
+                }
+            )
+
+            metadata.update(self._container.metadata)
+            metadata.update(self._video_stream.metadata)
+            return metadata
+
+        self._seek(index, constant_framerate=constant_framerate)
+        desired_frame = next(self._container.decode(video=0))
+
+        # useful flags defined on the frame
+        metadata.update(
+            {
+                "key_frame": bool(desired_frame.key_frame),
+                "time": desired_frame.time,
+                "interlaced_frame": bool(desired_frame.interlaced_frame),
```

Comment:
I always find it helpful if names that represent flags start with `is_`, e.g.  `is_interlaced` and `is_key_frame`. What do you think?

Reply:
Sure. This here is simply 1:1 plumbing. Inside FFmpeg these flags are called `key_frame` and `interlaced_frame`, so I figured it might be intuitive for users coming from that background to have the same names. Should we add a `is_` variant on top of this or as a replacement?

Reply:
If we do it, we should replace it. 

Good point - using the same terminology as ffmpeg is also worth something, so I'm fine if you leave it as-is.

Reply:
Then I will keep them as is, because my general understanding is to keep the `metadata` dict keys close to the names in the format/backend where possible.